File: manual/code/test_suites/cpp/FunctionBuildTest/tests/test_numpy_reshape.cpp
#include <gtest/gtest.h>
#include "../src/function_numpy_reshape.cpp"

TEST(ReshapeCalculatorTest, TestValidInput)
{
    ReshapeCalculator calc;
    std::string result = calc.reshape<double>(10, 5);
    EXPECT_NE(result.find("Price: 10"), std::string::npos);
    EXPECT_NE(result.find("Quantity: 5"), std::string::npos);
    EXPECT_NE(result.find("Total Amount: 50"), std::string::npos);
}

TEST(ReshapeCalculatorTest, TestZeroValue)
{
    ReshapeCalculator calc;
    std::string result = calc.reshape<double>(0, 5);
    EXPECT_NE(result.find("Price: 0"), std::string::npos);
    EXPECT_NE(result.find("Quantity: 5"), std::string::npos);
    EXPECT_NE(result.find("Total Amount: 0"), std::string::npos);
}

TEST(ReshapeCalculatorTest, TestNegativeValue)
{
    ReshapeCalculator calc;
    std::string result = calc.reshape<double>(-10, 5);
    EXPECT_NE(result.find("Price: -10"), std::string::npos);
    EXPECT_NE(result.find("Quantity: 5"), std::string::npos);
    EXPECT_NE(result.find("Total Amount: -50"), std::string::npos);
}

TEST(ReshapeCalculatorTest, TestFloatValue)
{
    ReshapeCalculator calc;
    std::string result = calc.reshape<double>(10.5, 3.2);
    EXPECT_NE(result.find("Price: 10.5"), std::string::npos);
    EXPECT_NE(result.find("Quantity: 3.2"), std::string::npos);
    EXPECT_NE(result.find("Total Amount: 33.6"), std::string::npos);
}

TEST(ReshapeCalculatorTest, TestDefaultValues)
{
    ReshapeCalculator calc;
    std::string result = calc.reshape<double>();
    EXPECT_NE(result.find("Price: 10"), std::string::npos);
    EXPECT_NE(result.find("Quantity: 5"), std::string::npos);
    EXPECT_NE(result.find("Total Amount: 50"), std::string::npos);
}

int main(int argc, char** argv)
{
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: manual/code/test_suites/python/python/test_numpy_reshape.py
import unittest
from function_numpy_reshape import ReshapeCalculator

class TestReshapeCalculator(unittest.TestCase):

    def test_valid_input(self):
        calc_reshape = ReshapeCalculator()
        result = calc_reshape.reshape(10, 5)
        self.assertIn("Price: 10", result)
        self.assertIn("Quantity: 5", result)
        self.assertIn("Total Amount: 50", result)

    def test_zero_value(self):
        calc_reshape = ReshapeCalculator()
        result = calc_reshape.reshape(0, 5)
        self.assertIn("Price: 0", result)
        self.assertIn("Quantity: 5", result)
        self.assertIn("Total Amount: 0", result)

    def test_negative_value(self):
        calc_reshape = ReshapeCalculator()
        result = calc_reshape.reshape(-10, 5)
        self.assertIn("Price: -10", result)
        self.assertIn("Quantity: 5", result)
        self.assertIn("Total Amount: -50", result)

    def test_float_value(self):
        calc_reshape = ReshapeCalculator()
        result = calc_reshape.reshape(10.5, 3.2)
        self.assertIn("Price: 10.5", result)
        self.assertIn("Quantity: 3.2", result)
        self.assertIn("Total Amount: 33.6", result)

    def test_default_values(self):
        calc_reshape = ReshapeCalculator()
        result = calc_reshape.reshape()
        self.assertIn("Price: 10", result)
        self.assertIn("Quantity: 5", result)
        self.assertIn("Total Amount: 50", result)

if __name__ == '__main__':
    unittest.main()
